### PR TITLE
Add a license field to the file metadata

### PR DESCRIPTION
**Pull request by Toflar**
File/Image licensing information feels like a very common requirement. Let's add this to the core.
My plan is to use that for JSON-LD once the whole response context stuff is settled :)

### CODE REVIEW

**Comment by m-vo** [comment]
Nice! Do we also want to output this by default in the image templates?

**Comment by fritzmg** [comment]
We actually decided against this once: https://github.com/contao/core-bundle/issues/1297#issuecomment-373392678

**Comment by Toflar** [comment]
In schema data then, yes. But not im HTML so :)

**Comment by fritzmg** [comment]
I am still very much in favour ;). Just wanted to mention it here. I think originally it was decided against adding a field by default in general, regardless of its output.

**Review by fritzmg — DISMISSED** [review]

**Comment by m-vo** [comment]
> We actually decided against this once: contao/core-bundle#1297 (comment)

Reading this, I think we maybe need some way to differentiate between/handle translated and (likely) non-translated properties like `license` at some point.

**Comment by Toflar** [comment]
I think we decided against it back then because there is no standardized way to display it in HTML. Hence we still don't do it. It's just preparation work for schema.org where it has a specific property :)

**Review by m-vo — DISMISSED** [review]

**Comment by fritzmg** [comment]
> I think we decided against it back then because there is no standardized way to display it in HTML.

There is though: https://github.com/contao/core-bundle/issues/1297#issuecomment-373394876

And I think some stock photo providers are pretty nitpicky anyway about where the license is displayed on the site. ([Apparently](https://community.contao.org/de/showthread.php?81070-Ursprüngliche-Metadaten-werden-im-Sourceset-entfernt) some even require you to have the meta data available in the image itself, which is currently not possible with Contao as these are stripped while processing the image.)

File: core-bundle/src/Resources/contao/dca/tl_files.php [hunk 1]
```diff
@@ -252,7 +252,8 @@
 					'title'           => 'maxlength="255"',
 					'alt'             => 'maxlength="255"',
 					'link'            => array('attributes'=>'maxlength="255"', 'dcaPicker'=>true),
-					'caption'         => array('type'=>'textarea')
+					'caption'         => array('type'=>'textarea'),
+					'license'         => array('attributes'=>'maxlength="255"', 'dcaPicker'=>true)
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/dca/tl_files.php:256]: Why did you add the picker here?

**Comment by ausi** [comment]
> ([Apparently](https://community.contao.org/de/showthread.php?81070-Urspr%C3%BCngliche-Metadaten-werden-im-Sourceset-entfernt) some even require you to have the meta data available in the image itself, which is currently not possible with Contao as these are stripped while processing the image.)

Related: https://github.com/contao/image/issues/80

**Comment by Toflar** [comment]
Most services support schema.org's [`ImageObject->license` property](https://schema.org/license), just like Google does. That's also why I added the picker because it's most likely to be an URL (which can also be a custom license page with your own terms).

File: core-bundle/src/Resources/contao/dca/tl_files.php (hunk 1 above)
Comment (Toflar) [reply core-bundle/src/Resources/contao/dca/tl_files.php:256]: https://github.com/contao/contao/pull/3076#issuecomment-858550272

**Comment by ausi** [comment]
> we maybe need some way to differentiate between/handle translated and (likely) non-translated properties like `license` at some point.

How about adding a feature to the `metaWizard` so that it can handle non-language specific fields too? Or how about just adding it as a regular column to `tl_files`?

**Comment by Toflar** [comment]
`license` is a translated property though. So that's for a different discussion and once again, not related to this PR.

**Review by ausi — DISMISSED** [review]

**Comment by leofeyer** [comment]
> So that's for a different discussion and once again, not related to this PR.

I think the discussion is related to this PR, because you are now adding the field as translated property, which e.g. does not make much sense if the entry is always `https://creativecommons.org/licenses/by/4.0/`.

**Comment by Toflar** [comment]
If it is the same URL, then yes. But you can have the same license description in many languages. If there's only one, then only add it to your fallback language.

**Comment by fritzmg** [comment]
> If there's only one, then only add it to your fallback language.

But all the meta fields are always translated together, not individually, aren't they? So you cannot have a fallback just for the `license` field at the moment, whereas `title` should still be translated for example.

**Comment by m-vo** [comment]
Yes, currently the first locale that has data wins. But we could maybe adjust this. Probably not without affecting BC, though.

**Comment by ausi** [comment]
> I think the discussion is related to this PR, because you are now adding the field as translated property, which e.g. does not make much sense if the entry is always `https://creativecommons.org/licenses/by/4.0/`.

The `link` property might also not need to be translated in many cases, so I think the `license` field could be seen the same way as the current `link` field.

> If there's only one, then only add it to your fallback language.

As the fallback language is defined per root page this is probably problematic. But we could extend the `metaWizard` to support entering a fallback explicitly. But that would be not related to this PR 😊

**Review by ausi — APPROVED** [review]

**Review by fritzmg — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.
